docker/build-tools/prow-entrypoint.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eux
+
+# Authenticate gcloud, allow failures
+if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
+  gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}" || true
+fi
+
+. dockerd-entrypoint.sh
```

Comment:
didnt have a chance to look into this script, will this unconditionally run `dockerd` (and I assume some other stuff as well)?

Reply:
the script is fairly difficult to follow.  The logic looks like:
```
# no arguments passed
# or first arg is `-f` or `--some-option`
if [ "$#" -eq 0 ] || [ "${1#-}" != "$1" ]; then
    ... do some setup of dockerd
    run dockerd
else
if $1=dockerd
  run rootlesskit or dind depending on arguments
else
   run docker-entrypoint.sh - which does some additional logic.. :)
fi
```

I'm really not certain after reading the bash - i was hoping to have a test in ci.

Cheers
-steve

Reply:
It looks to me like the `docker:dind` image from the docs is meant to be run as part of a pod, rather than individually.  On startup, the `docker:dind` image will run dockerd.  Then an application can connect to dockerd via the network...

Wiill need to give this some more thought.  Just running dockerd seems simpler than messing around with pods - especially on a laptop :)